### PR TITLE
jsonschema2pojo 0.4.13

### PR DESCRIPTION
**Pull request by kikuchy**

### CODE REVIEW

**Comment by dunn** [comment]
Thanks for the submission! The bot flagged two problems: the description needs to be a bit shorter (80 characters or less, including the name of the formula), and the test isn't currently passing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/29932/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_jsonschema2pojo/

**Comment by kikuchy** [comment]
Thank you for your advice!
But I'm not good at ruby, I can't find the way to test the Formula.

`brew test jsonshema2pojo` on my mac passed the test.
What can I do for it?

**Comment by kikuchy** [comment]
Thank you Kerosene2000!
I just overlooked separation with commas X(

Now I fixed arguments of `system` and `File.exist?`.

**Comment by kikuchy** [comment]
@dunn Excuse me, when this request will merge?

File: Library/Formula/jsonschema2pojo.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    Dir.mktmpdir do |outdir|
+      Dir.mktmpdir do |srcdir|
+        f = File.open("#{srcdir}/jsonschema.json", "w")
+        f.write json
+        f.close
+        system "#{bin}/jsonschema2pojo", "-s", f.path, "-t", outdir
+        File.exist? "#{outdir}/Jsonschema.java"
+      end
+    end
```
Comment (bfontaine) [inline Library/Formula/jsonschema2pojo.rb]: Why is it needed to create two temporary directories? Can’t you test everything in the build path?

File: Library/Formula/jsonschema2pojo.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        }
+      }
+    }
+    EOS
+
+    Dir.mktmpdir do |outdir|
+      Dir.mktmpdir do |srcdir|
+        f = File.open("#{srcdir}/jsonschema.json", "w")
+        f.write json
+        f.close
```
Comment (bfontaine) [inline Library/Formula/jsonschema2pojo.rb]: Use `(srcdir/"jsonschema.json").write json` here

**Comment by DomT4** [comment]
Ping.

**Comment by bfontaine** [comment]
Ping?

**Comment by kikuchy** [comment]
Sorry, I'm been busy...
It's hard to take my time to fix it for the being.

> Can’t you test everything in the build path?

Can I use `buildpath` in `test` block?
I tried to print it, but only `nil` was printed.


> `(srcdir/"jsonschema.json").write json` 

I'm not good at Ruby. So I can't understand it.
Is it valid syntax?
In my environment, it cause an error of "undefined method".

```
Testing jsonschema2pojo
Error: jsonschema2pojo: failed
undefined method `/' for #<String:0x007fd90c13e1c8>
/usr/local/Library/Formula/jsonschema2pojo.rb:37:in `block (3 levels) in <class:Jsonschema2pojo>'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/tmpdir.rb:88:in `mktmpdir'
/usr/local/Library/Formula/jsonschema2pojo.rb:36:in `block (2 levels) in <class:Jsonschema2pojo>'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/tmpdir.rb:88:in `mktmpdir'
/usr/local/Library/Formula/jsonschema2pojo.rb:35:in `block in <class:Jsonschema2pojo>'
/usr/local/Library/Homebrew/formula.rb:811:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:17:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:807:in `run_test'
/usr/local/Library/Homebrew/test.rb:28:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/usr/local/Library/Homebrew/test.rb:27:in `<main>'
```

File: Library/Formula/jsonschema2pojo.rb (hunk 2 above)
Comment (bfontaine) [inline Library/Formula/jsonschema2pojo.rb]: `srcdir` is a String so it won’t work. You can use the following instead:

```rb
File.open("#{srcdir}/jsonschema.json", "w") do |f|
  f.write json
end
```

File: Library/Formula/jsonschema2pojo.rb (hunk 1 above)
Comment (bfontaine) [inline Library/Formula/jsonschema2pojo.rb]: Sorry it should be “test path”, not “build path”. Would the following test work?

```rb
  test do
    json = <<-EOS.undent
    {
      "type":"object",
      "properties": {
        "foo": {
          "type": "string"
        },
        "bar": {
          "type": "integer"
        },
        "baz": {
          "type": "boolean"
        }
      }
    }
    EOS

    (buildpath/"jsonschema.json").write json
    system "#{bin}/jsonschema2pojo", "-s", testpath, "-t", testpath
    assert (buildpath/"Jsonschema.java").exist?
  end
```

**Comment by kikuchy** [comment]
@bfontaine

Thank you!
I can use `testpath` and make simple test code.

**Comment by bfontaine** [comment]
Thank you for your contribution @kikuchy; We appreciate it! :tada:

**Comment by kikuchy** [comment]
Thanks a lot!! 🍺

**Pull request closed without merging**